plugins/imagebase/plugin.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or http://ckeditor.com/license
+ */
+
+( function() {
+	'use strict';
+
+	function createWidgetDefinition( editor, definition ) {
+		var defaultTemplate = new CKEDITOR.template(
+			'<figure class="{basicClass}">' +
+				'<img alt="" src="" />' +
+				'<figcaption>{captionPlaceholder}</figcaption>' +
+			'</figure>' ),
+			baseDefinition;
+
+		/**
+		 * This is an abstract class that describes a definition of a basic image widget
+		 * creted by {@link CKEDITOR.plugins.imagebase#addImageWidget} method.
+		 *
+		 * Note that because the image widget is a type of a widget, this definition extends
+		 * {@link CKEDITOR.plugins.widget.definition}.
+		 * It adds several parts of image and implements the basic version of
+		 * {@link CKEDITOR.plugins.widget.definition#upcast} callback.
+		 *
+		 * @abstract
+		 * @since 4.8.0
+		 * @class CKEDITOR.plugins.imagebase.imageWidgetDefinition
+		 * @mixins CKEDITOR.plugins.widget.definition
+		 */
+		baseDefinition = {
+			pathName: editor.lang.imagebase.pathName,
+
+			template: defaultTemplate,
+
+			allowedContent: {
+				img: {
+					attributes: '!src,alt,width,height'
+				},
+				figure: true,
+				figcaption: true
+			},
+
+			requiredContent: 'figure',
```

Comment:
`img[src]` should be required too, as there's pretty much no point in putting figure alone.